Description: Detect KFreeBSD
 Seems that __FreeBSD__ preprocessor macro is not provided by KFreeBSD.
 Add an ad-hoc test for it and declare it as UNIX
Author: Agustin Martin Domingo <agmartin@debian.org>
Last-Update: 2016-12-09

--- encfs-1.9.1.orig/internal/easylogging++.h
+++ encfs-1.9.1/internal/easylogging++.h
@@ -99,13 +99,18 @@
 #else
 #   define ELPP_OS_FREEBSD 0
 #endif
+#if (defined(__FreeBSD_kernel__) && defined(__GLIBC__))
+#   define ELPP_OS_KFREEBSD 1
+#else
+#   define ELPP_OS_KFREEBSD 0
+#endif
 #if (defined(__sun))
 #   define ELPP_OS_SOLARIS 1
 #else
 #   define ELPP_OS_SOLARIS 0
 #endif
 // Unix
-#if ((ELPP_OS_LINUX || ELPP_OS_MAC || ELPP_OS_FREEBSD || ELPP_OS_SOLARIS) && (!ELPP_OS_WINDOWS))
+#if ((ELPP_OS_LINUX || ELPP_OS_MAC || ELPP_OS_FREEBSD || ELPP_OS_KFREEBSD || ELPP_OS_SOLARIS) && (!ELPP_OS_WINDOWS))
 #   define ELPP_OS_UNIX 1
 #else
 #   define ELPP_OS_UNIX 0
